Calendar aligned timeframe feature (#20553)
DataDog/documentation: content/en/dashboards/guide/custom_time_frames.md

kind: guide
---

## Overview

> <div class="alert alert-info">Queries are run in UTC time, but the query time range is selected according to your <strong>browser</strong>. Toggle between displaying the default time zone or UTC from the dashboard configure action. For more information, see the <a href="/dashboards/#display-utc-time">Dashboard Settings documentation</a>.</div>

Reviewer: ```suggestion
<div class="alert alert-info">Queries are run in UTC time, but the query time frame is selected according to your <strong>browser's time zone</strong>. Toggle between displaying the default time zone or UTC from the dashboard configure action. For more information, see the <a href="/dashboards/#display-utc-time">Dashboard Settings documentation</a>.</div>
```

- Time range -> time frame to be consistent with the rest of the page.
- browser -> browser's time zone (I'm assuming that's what it means?)

{{< img src="dashboards/guide/custom_time_frames/custom_relative_time_frame.mp4" alt="Type custom relative time frame" video="true" width="500" >}}

> **Note**: Queries are run in UTC time, but the query time range is selected according to your browser. Toggle between displaying the default time zone or UTC from the dashboard configure action. For more information, see the [Dashboard Settings][1] documentation.

Reviewer 2: I would try to highlight (in bold for example) the fact that the timezone should be the user's browser.

| Unix seconds timestamp       | 1577883600                                       |
| Unix milliseconds timestamp  | 1577883600000                                    |

Any fixed date can be entered as part of a range. For example:
  * `1577883600 - 1578009540`
  * `Jan 1 - Jan 2`
  * `6:00 am - 1:00 pm`

### Relative dates

> Relative dates **do not** update as time passes, they are calculated when entered.

Reviewer: ```suggestion
Relative dates **do not** update over time; they are calculated when entered.
```
Change `,` to `;` to fix comma splice.

|----------------|----------------------------------------------------------------------------------|
| `N{unit}`<br> See the list of accepted units below      | Displays the past N units. For example, **3 mo** (the past 3 months)             |
| `today`        | Displays the current calendar day until present                                 |
| `yesterday`    | Displays the full previous calendar day                                         |
| `this month`   | Displays the current calendar month until present                                |
| `last month`   | Displays the full previous calendar month                                        |
| `this year`    | Displays the current calendar year until present                                 |
| `last year`    | Displays the full previous calendar year                                         |

The following strings are accepted for any `{unit}` in a relative date:
  * Minutes: `m`, `min`, `mins`, `minute`, `minutes`
  * Hours: `h`, `hr`, `hrs`, `hour`, `hours`
  * Days: `d`, `day`, `days`
  * Weeks: `w`, `week`, `weeks`
  * Months: `mo`, `mos`, `mon`, `mons`, `month`, `months`

### Calendar aligned dates

> Calendar aligned dates update to reflect the current present day. For example, if you select `week to date` on November 2, on November 9, it displays data from November 1 - November 9.

Reviewer: ```suggestion
Calendar aligned dates update to reflect the current day. For example, if you select `week to date` on November 2, on November 9, it displays data from November 1 - November 9.
```
I think "current present" is probably redundant.

|----------------|----------------------------------------------------------------------------------|-----------------------------------------------------------|
| `N{unit}`      | 3m<br>3 min<br>3h<br>3 hours<br>3d<br>3 days<br>3w<br>3 weeks<br>3mo<br>3 months | Displays the past N units, for example: the past 3 months |
| `today`        | Nov 9, 12:00 am - Nov 9, 3:15 pm                                                 | Displays the current calendar day until present           |
| `yesterday`    | Nov 8, 12:00 am - Nov 9, 12:00 am                                                | Displays the full previous calendar day                   |
| `this month`   |                                                                                  | Displays the current calendar month until present         |
| `last month`   |                                                                                  | Displays the full previous calendar month                 |
| `this year`    |                                                                                  | Displays the current calendar year until present          |

> | `last year`    |                                                                                  | Displays the full previous calendar year                  |

Reviewer 2: Is it expected not to have any examples for this month, last month, this year, last year?

Author: There weren't any listed, but we can add them if you think it would be helpful!

Reviewer 2: I think it would make sense, it felt weird reading the table where some fields are blank
